Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp

```diff
@@ -8,121 +8,141 @@
 
-  // output of function needed for seed
-
-  finder.loadSpacePoints(GbtsSpacePoints);
+  // used to reserve size of nodes 2D vector in core
+  int max_layers = m_LayeridMap.size();
 
-  // trigGbts file :
-  Acts::Experimental::RoiDescriptor internalRoi(
-      0, -4.5, 4.5, 0, -std::numbers::pi, std::numbers::pi, 0, -150., 150.);
-  // ROI file:
+  // ROI file:Defines what region in detector we are interested in, currently
+  // set to entire detector
   //  Acts::Experimental::RoiDescriptor internalRoi(0, -5, 5, 0,
   //  -std::numbers::pi, std::numbers::pi, 0, -225., 225.);
+  Acts::Experimental::RoiDescriptor internalRoi(
+      0, -4.5, 4.5, 0, -std::numbers::pi, std::numbers::pi, 0, -150., 150.);
 
-  // new version returns seeds
-  SimSeedContainer seeds = finder.createSeeds(internalRoi, *m_gbtsGeo);
+  // create the seeds
+
+  Acts::SeedContainer2 seeds =
+      finder.CreateSeeds(internalRoi, SpContainerComponents, max_layers);
+
+  // move seeds to simseedcontainer to be used down stream taking fist middle
+  // and last sps currently as simseeds need to be hard types so only 3
+  // spacepoint can be added but in future we should be able to have any length
+  // seed
+  SimSeedContainer seedContainerForStorage;
+  seedContainerForStorage.reserve(seeds.size());
+  for (const auto &seed : seeds) {
+    auto sps = seed.spacePointIndices();
+    unsigned int indices = sps.size() - 1;
+    std::size_t mid = static_cast<std::size_t>(std::round(indices / 2.0));
+    seedContainerForStorage.emplace_back(
+        *std::get<0>(SpContainerComponents)
+             .at(sps[0])  // first spacepoint
+             .sourceLinks()[0]
+             .get<const SimSpacePoint *>(),
+        *std::get<0>(SpContainerComponents)
+             .at(sps[mid])  // middle spacepoint
+             .sourceLinks()[0]
+             .get<const SimSpacePoint *>(),
+        *std::get<0>(SpContainerComponents)
+             .at(sps[indices])  // last spacepoint
+             .sourceLinks()[0]
+             .get<const SimSpacePoint *>());
+
+    seedContainerForStorage.back().setVertexZ(seed.vertexZ());
+    seedContainerForStorage.back().setQuality(seed.quality());
+  }
 
-  m_outputSeeds(ctx, std::move(seeds));
+  m_outputSeeds(ctx, std::move(seedContainerForStorage));
 
   return ActsExamples::ProcessCode::SUCCESS;
 }
 
-std::map<std::pair<int, int>, std::pair<int, int>>
+std::map<std::pair<int, int>, std::tuple<int, int, int>>
 ActsExamples::GbtsSeedingAlgorithm::makeActsGbtsMap() const {
-  std::map<std::pair<int, int>, std::pair<int, int>> ActsGbts;
+  std::map<std::pair<int, int>, std::tuple<int, int, int>> ActsGbts;
+
+  // prepare the acts to gbts mapping file
   std::ifstream data(
       m_cfg.layerMappingFile);  // 0 in this file refers to no Gbts ID
   std::string line;
-  std::vector<std::vector<std::string>> parsedCsv;
+  std::vector<std::vector<std::string>>
+      parsedCsv;  // row = physical module, column = ACTS ID components
```

Comment:
the comments at the end of the line result in artificial line breaks. to me it seems more readable to move the comments above the individual lines